src/psfmachine/machine.py

```diff
@@ -1132,3 +1245,38 @@ def fit_model(self, fit_va=False):
             self.werrs[:, nodata] *= np.nan
 
         return
+
+
+def sparse_lessthan(arr, limit):
+    """
+    Compute less than operation on sparse array by evaluating only non-zero values
+    and reconstructing the sparse array.
+
```

Comment:
Can you add a quick comment explaining why we need a function for this, and how this function works in a few words?

Reply:
might be a comment in the other file, you could move that here